tests/ContextTest.py

```diff
@@ -97,3 +100,86 @@ def test_add_import(self):
         ctx3.add_import(ctx)
         final_ctx = Context(ident='http://example.com/context_1', imported=(ctx3,))
         self.assertEqual(len(final_ctx.save_imports()), 4)
+
+    def test_init_len(self):
+        ctx = Context(ident='http://example.com/context_1')
+        self.assertEqual(len(ctx), 0)
+
+    def test_len(self):
+        ident_uri = 'http://example.com/context_1'
+        ctx = Context(ident=ident_uri)
+        for i in range(5):
+            ctx.add_statement(create_mock_statement(ident_uri, i))
+        self.assertEqual(len(ctx), 5)
+
+    def test_add_remove_statement(self):
+        ident_uri = 'http://example.com/context_1'
+        ctx = Context(ident=ident_uri)
+        stmt_to_remove = create_mock_statement(ident_uri, 42)
+        for i in range(5):
+            ctx.add_statement(create_mock_statement(ident_uri, i))
+        ctx.add_statement(stmt_to_remove)
+        ctx.remove_statement(stmt_to_remove)
+        self.assertEqual(len(ctx), 5)
+
+    def test_add_statement_with_different_context(self):
+        ctx = Context(ident='http://example.com/context_1')
+        stmt1 = create_mock_statement('http://example.com/context_2', 1)
+        with self.assertRaises(ValueError):
+            ctx.add_statement(stmt1)
+
+    def test_contents_triples(self):
+        res_wanted = []
+        ident_uri = 'http://example.com/context_1'
+        ctx = Context(ident=ident_uri)
+        for i in range(5):
+            stmt = create_mock_statement(ident_uri, i)
+            ctx.add_statement(stmt)
+            res_wanted.append(stmt.to_triple())
+        self.assertEqual(list(ctx.contents_triples()), res_wanted)
```

Comment:
These should be compared without respect to order. The fact that `contents_triples()` returns them in the order they were added should be treated as an implementation detail, so it's not something I want to test.